### ISSUE TITLE
Fix data leakage in XGBoost baseline (LabelEncoder fit on train+test)

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/cruijff_kit#428]
## Context

While reviewing the codebase for ML data leakage — prompted by reading "Leakage and the Reproducibility Crisis in ML-based Science" (Kapoor & Narayanan, 2023) — we found a preprocessing leak in the XGBoost baseline.

## Problem

In `experiments/folktexts/ml_baseline.py` (lines 160-163), `LabelEncoder` is fit on the union of train and test data:

```python
for col in categorical_cols:
    encoders[col] = LabelEncoder()
    all_values = [ex.get(col, "") for ex in train_raw + test_raw]  # leakage
    encoders[col].fit(all_values)
```

This means the encoder sees categorical values from the test set during training. If the test set contains category values absent from training, the encoder silently accommodates them rather than treating them as unknown — inflating XGBoost's reported performance.

Note: the CatBoost baseline in the same file (`prepare_catboost_data()`) does **not** have this issue since CatBoost handles categoricals natively.

## Proposed Fix

Fit encoders on `train_raw` only and handle unseen test values gracefully. Two options:

1. **Manual dict encoding** with a fallback value (e.g., -1) for unknowns — simple, minimal change
2. **`OrdinalEncoder`** with `handle_unknown="use_encoded_value", unknown_value=-1` — sklearn-native

## Additional Findings

During the same review, two non-leakage issues were noted (not blocking, but worth tracking):

- **Missing random seed** in `experiments/capitalization/preprocess_input_data.py:93` — `random.sample()` without `random.seed()`, making dataset generation non-reproducible. Other generators (synthetic_twins, bit_sequences) correctly set seeds.
- **Split name mismatch** in `sanity_checks/bit_sequences/` — generator creates `train`/`validation` splits but the inspect task defaults to `split: "test"`. Would cause a runtime error, not leakage.

**Comment by niznik-dev** [comment]
Not urgently concerned about this one (since us using XGBoost is a sidequest) but wanted to get it noted before moving on to other thoughts
